rasa/core/actions/forms.py

```diff
@@ -55,7 +55,14 @@ def required_slots(self, domain: Domain) -> List[Text]:
         Returns:
             A list of slot names.
         """
-        return list(domain.slot_mapping_for_form(self.name()).keys())
+        form = domain.slot_mapping_for_form(self.name())
```

Comment:
I would handle the `required_slots` stuff within the domain and consolidate it there.
I'd imagine the workflow like this:

1. `Domain` is loaded from YAML
2. `Domain.__init__` double checks form format. In case the users didn't use `required_slots` it adjusts it so that the dictionary looks like `{"forms": {"my_form": {"required_slots": {"my_slot": [] }}}}`